src/GoogleFitOnFhir/Services/GoogleFitAuthService.cs

```diff
@@ -0,0 +1,65 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using GoogleFitOnFhir.Clients.GoogleFit;
+using GoogleFitOnFhir.Clients.GoogleFit.Responses;
+using Microsoft.Extensions.Logging;
+
+namespace GoogleFitOnFhir.Services
+{
+    public class GoogleFitAuthService : IGoogleFitAuthService
+    {
+        private readonly ILogger<GoogleFitAuthService> _logger;
+        private readonly GoogleFitClientContext _clientContext;
+        private readonly IGoogleFitAuthUriRequest _googleFitAuthUriRequest;
+        private readonly IGoogleFitAuthTokensRequest _googleFitAuthTokensRequest;
+        private readonly IGoogleFitRefreshTokenRequest _googleFitRefreshTokensRequest;
```

Comment:
I am starting to feel like these 3 classes (GoogleFitAuthTokensRequest, GoogleFitAuthUriRequest, and GoogleFitRefreshTokensRequest are adding an unneeded layer go abstraction that is causing code bloat and making testing difficult.

This IGoogleFitAuthService already serves as a layer of abstraction for the Google endpoints. I think we can move away from these request classes and simply "new up" the instance of GoogleAuthorizationCodeFlow in the constructor of this class, make the 3 calls represented by the request classes directly on the GoogleAuthorizationCodeFlow object and convert the responses to the correct return types.

This would mean that this class is not testable, but I'm fine with that because this is just a wrapper for the Google APIS.